modules/core/src/main/java/org/apache/ignite/internal/util/CollectionUtils.java

```diff
@@ -521,4 +523,16 @@ public static <T> Set<T> intersect(Set<T> op1, Set<T> op2) {
 
         return Set.copyOf(set);
     }
+
+    /**
+     * Gets a list based on an iterator.
+     *
+     * @param iterator The iterator to use.
+     * @return A list of the iterator contents.
+     * @throws NullPointerException if iterator parameter is null
+     */
+    public static <T> List<T> toList(Iterator<T> iterator) {
```

Comment:
As I see the conversion required just for tests. If it so it shouldn't be here. For tests exists org.junit.platform.commons.util.CollectionUtils.toStream and so one.

Reply:
This can be used by the customer to process the criteria query results, but let the customer choose what to use.
`org.junit.platform.commons.util.CollectionUtils.toStream` - Any usage by external parties is not supported
So I replaced it with `Lists#newArrayList`, which was available by default for tests

Reply:
It can't be used by customer due to it is internal utilities.